src/main/xerus/monstercat/api/Playlist.kt

```diff
@@ -86,4 +101,181 @@ object Playlist {
 			else add(track)
 		}
 	}
+}
+
+object PlaylistManager {
+	suspend fun loadPlaylist(apiConnection: APIConnection){
+		val tracks = apiConnection.getTracks()
+		tracks?.forEachIndexed { index, track ->
+			val found = Cache.getTracks().find {
+				it.id == track.id
+			}
+			if (found != null)
+				tracks[index] = found
+			else
+				tracks.removeAt(index)
+		}
+		if (tracks != null && tracks.isNotEmpty()) {
+			if (Player.player?.status != MediaPlayer.Status.DISPOSED)
+				Player.reset()
+			Playlist.setTracks(tracks)
+		}
+	}
+
+	/** Opens the playlist manager dialog
+	 * Allows to load, save, and manage playlists stored on Monstercat.com
+	 */
+	fun playlistDialog(){
+		val connection = APIConnection("api", "playlist").fields(ConnectPlaylist::class)
+
+		val parent = VBox()
+		val stage = App.stage.createStage("Monstercat.com Playlists", parent)
+		stage.initModality(Modality.WINDOW_MODAL)
+		val connectTable = TableView<ConnectPlaylist>()
+
+		// Common playlist functions
+		fun load(runAfter: () -> Unit = {}) {
+			if (connectTable.selectionModel.selectedItem != null) {
+				GlobalScope.async {
+					loadPlaylist(APIConnection("api", "playlist", connectTable.selectionModel.selectedItem.id, "tracks"))
+					onFx { runAfter.invoke() }
+				}
+			}
+		}
+		fun loadUrl(runAfter: () -> Unit = {}) {
+			val subParent = VBox()
+			val subStage = stage.createStage("Load from URL", subParent)
+			subStage.initModality(Modality.WINDOW_MODAL)
+			val textField = TextField().apply { promptText = "URL" }
+			subParent.add(textField)
+			subParent.addRow(createButton("Load"){
+				val playlistId = textField.text.substringAfterLast("/")
+				if (playlistId.length == 24){
+					GlobalScope.async {
```

Comment:
Why async and not launch?

Reply:
What's the difference ?

Reply:
I replaced it 0f6dd4f179479101a8ea5a094d612b9ef62cfec0 but can you explain the difference either here or in discord ?